src/gf.c

```diff
@@ -2636,7 +2636,8 @@ jl_code_instance_t *jl_compile_method_internal(jl_method_instance_t *mi, size_t
                     codeinst->rettype_const = unspec->rettype_const;
                     jl_atomic_store_release(&codeinst->invoke, unspec_invoke);
                     jl_mi_cache_insert(mi, codeinst);
-                    record_precompile_statement(mi, 0);
+                    if (!def->is_for_opaque_closure)
```

Comment:
Can you also add a check for `codeinst-> owner === jl_nothing`

Seems like I forgot about that.

Reply:
Sure thing - is there a test I can add so that we don't regress?